woocommerce/woocommerce-android
# Variations: No price warning banners

Fixes #2680. 

The PR introduces warning banners in variation list or below the price property in variation detail if a variation is set as enabled but does not have price. 

The variation list item labels have also been updated from **Visible/Hidden** -> **Enabled/Disabled**.


| Light | Dark |
| - | - |
| ![image](https://user-images.githubusercontent.com/1522856/89878873-70c02380-dbc2-11ea-9e8b-4ec1869a1a54.png) | ![image](https://user-images.githubusercontent.com/1522856/89879011-99481d80-dbc2-11ea-8987-b1cf019fd476.png) | 
| ![image](https://user-images.githubusercontent.com/1522856/89879077-aa912a00-dbc2-11ea-85a9-8bbc4016df92.png) | ![image](https://user-images.githubusercontent.com/1522856/89879099-b4b32880-dbc2-11ea-831c-301808dced45.png) |
| ![image](https://user-images.githubusercontent.com/1522856/89879251-f04df280-dbc2-11ea-8022-d327092fdd40.png)

**To test**

1. Open a variable product
2. Tap on Variations property
3. Tap an a variation
4. Remove the price (if set)
5. Enable the variation (if not enabled)
6. Notice the warning label is displayed
7. Save the variation
8. Go back to the list
9. Notice a warning is displayed above the list & the "No price set" label is displayed for the variation item
10. Open the same variation detail again
11. Set the price
12. Notice the warning disappears
13. Save the variation
14. Go back to the list
15. Notice there is no warning displayed (if there is no other enabled variation without price)
